### ISSUE TITLE
Mancy GUI never appears

### ISSUE THREAD

**Issue by clementi** [princejwesley/Mancy#73]
After upgrading to 2.1.0 on Windows 7 Pro, the REPL GUI does not appear when I start Mancy. I have repeatedly removed and re-downloaded Mancy to no effect. I have also deleted %appdata%\mancy to no effect. If I delete the 2.1.0 installation and run the 2.0.1 build, the GUI starts up.

**Comment by princejwesley** [comment]
@clementi  I don't have access to windows 7 machine :disappointed: . I am not sure what went wrong. @senthilporunan tested in windows 7 before release tag is created.

 is it 32 bit or 64 bit machine? Can you clone master branch or 2.1.0 tag and build the exe?

**Comment by clementi** [comment]
@princejwesley It's 64 bit. I'll clone and try a build.

**Comment by clementi** [comment]
I'm unable to get a build working: electron-rebuild is failing.

**Comment by princejwesley** [comment]
@clementi electron-prebuilt is installed ?

**Comment by clementi** [comment]
@princejwesley electron-prebuilt 0.35.4 appears to be installed:

```
F:\projects\Mancy>npm list | grep electron
├─┬ electron-packager@5.1.0
│ ├─┬ electron-download@1.4.1
├── electron-prebuilt@0.35.4
├─┬ electron-rebuild@1.0.2
```

**Comment by princejwesley** [comment]
@clementi I am looking at the electron-rebuild [source code](https://github.com/electronjs/electron-rebuild/blob/master/src/main.js#L42).

can you run the following and check the results?
```
>  cat node_modules/electron-prebuilt/path.txt
/Users/princejohnwesley/Projects/Playground/Node/electron/Mancy/node_modules/electron-prebuilt/dist/Electron.app/Contents/MacOS/Electron

> ATOM_SHELL_INTERNAL_RUN_AS_NODE=1 /Users/princejohnwesley/Projects/Playground/Node/electron/Mancy/node_modules/electron-prebuilt/dist/Electron.app/Contents/MacOS/Electron -e 'console.log(process.versions.modules)'
46
```

As per your logs, looks like you are getting `(blank line)46`. If that is true, [this](https://github.com/electronjs/electron-rebuild/blob/master/src/main.js#L49) line requires a fix.

Try modifying `node_modules/electron-rebuild/src/main.js:48` as below and run:
```js
  if (!versionAsString.trim().match(/^\d+$/)) {
    throw new Error(`Failed to check Electron's module version number: ${versionAsString}`);
  }
```

**Comment by princejwesley** [comment]
@clementi  There is an [open ticket](https://github.com/electronjs/electron-rebuild/issues/37) for this issue. As per your log, I suspect that stdout is leaking

**Comment by clementi** [comment]
@princejwesley I confirm 46.

**Comment by clementi** [comment]
I wonder how @senthilporunan was able to build?

**Comment by princejwesley** [comment]
@clementi You can pass in electron-prebuilt path explicitly [like this](https://github.com/electronjs/electron-rebuild/blob/master/src/cli.js#L23). 

Try giving
```
>    "postinstall": "electron-rebuild",
```
to something like this
```
>     "postinstall": "electron-rebuild e  electron-path",
```

**Comment by princejwesley** [comment]
@clementi I built it from windows 10, 32 bit machine (64 bit binary) and asked him to test. He was able to run the executable.

**Comment by princejwesley** [comment]
@clementi For time being, set `versionAsString`([here](https://github.com/electronjs/electron-rebuild/blob/master/src/main.js#L49)) as 46 and build.

**Comment by clementi** [comment]
@princejwesley Setting versionAsString to `"46"` has no effect. Neither does passing any path to electron-rebuild. However, when I run `npm run package` I get an (incomplete) build of Mancy and when I run it, the GUI _does_ show up.

Hmm. I'm using Node 4.x.x. I'll try 5.x.x.

**Comment by princejwesley** [comment]
Electron rebuild is not required if installed node version and electron backed node versions are same.

**Comment by ljgww** [comment]
It does create 3 Mancy.exe processes though.
My setup default node 4.2.1 but I got impression that electron uses its own node 4.1.1

**Comment by princejwesley** [comment]
@ljgww Yes, thats why we need to rebuild native modules. are you facing similar issue ?

**Comment by princejwesley** [comment]
@ljgww @clementi use of `await` in [electron-rebuild](https://github.com/electronjs/electron-rebuild/blob/master/src/main.js#L46) is buggy. thats why `stdout` is leaking to parent process.

**Comment by AddictArts** [comment]
Here is some help I hope. Fresh Windows 7 x64 machine. I tried both nodejs 4.4 and 5.8 and it did not matter which one for me, as well as npm 2 vs npm 3. So, now that that is out the way.

* I installed nodejs using the msi
* I did a git clone first.

1. font-manager would failed to build. So, to fix it I bumped the version to 0.2.2 in package.json

npm install worked after that.

2, npm run package

ran without failure. But, running the executable has an issue. The window opens but nothing appears in the main part of the window as though the render is not happening.

3. npm run debug and npm start

ran without failure and the window and html part render just fine. This worked, so I made a bat file to launch it and shortcut for now. I didn't have time to look into why the exe seems to not be rendering.

**Comment by princejwesley** [comment]
@AddictArts font-manager 0.2.2 version is a new issue due to electron upgrade. I have a fix in clojurescript branch. I'll merge clojurescript branch to master in few days. Sorry for the incovinence
